src/lib/y2firstboot/clients/licenses.rb

```diff
@@ -0,0 +1,87 @@
+# encoding: utf-8
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+Yast.import "UI"
+Yast.import "Misc"
+Yast.import "GetInstArgs"
+
+module Y2Firstboot
+  module Clients
+    # Client that allows to display license texts during the firstboot
+    # configuration.
+    #
+    # NOTE: for backward compatibility, it can display the Novell licenses if a
+    # valid path is given through FIRSTBOOT_NOVELL_LICENCE_DIR
+    class Licenses < Yast::Client
+      def initialize
+        textdomain "firstboot"
+      end
+
+      def run
+        result = WFM.CallFunction("inst_license", inst_license_args)
+
+        UI.CloseDialog if result == :halt
```

Comment:
hmm, so every client have to close himself dialog if result is halt? cannot be done one level high?

Reply:
Honestly, I don't know.

I tried to clean the client as much as possible, but this line I was not sure if could be changed. Maybe the firstboot module is only being closed if the user refuse the license.

Reply:
You cannot change it unless it is changed globally. So keep as it is now, maybe something to improve for future.